spec/controllers/recordings_controller_spec.rb

```diff
@@ -53,6 +53,46 @@
     end
   end
 
+  describe '#update' do
+    # Metaprogramming -> mocking underlying method implemntation on the run.
+
+    let(:recording_update_params) do
+      {
+        name: 'My Awesome Recording!'
+      }
+    end
+
+    let(:invalid_recording_update_params) do
+      {
+        name: ''
+      }
+    end
+
+    let(:room) { create(:room, user:) }
+    let(:recording) { create(:recording, room:) }
+
+    before do
+      allow_any_instance_of(BigBlueButtonApi).to receive(:update_recordings).and_return(http_ok_response)
+    end
+
+    it 'updates the recordings name with valid params returning :ok status code' do
+      expect { post :update, params: { recording: recording_update_params, id: recording.record_id } }.to(change { recording.reload.name })
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'does not update the recordings name for invalid params returning a :bad_request status code' do
+      expect { post :update, params: { recording: invalid_recording_update_params, id: recording.record_id } }.not_to(change do
+                                                                                                                        recording.reload.name
+                                                                                                                      end)
```

Comment:
This indentation is weird.

```
expect { 
  post :update, 
  params: { recording: invalid_recording_update_params, id: recording.record_id } 
}.not_to change { recording.reload.name }
```

Reply:
I'm using an extension to correct my work according to our Rubocop config.
What you have provided Rubocop doesn't seem to like.
Do you approve this instead:
```
     expect do
        post :update,
             params: { recording: invalid_recording_update_params, id: recording.record_id }
      end.not_to(change { recording.reload.name })
```